src/serde.rs

```diff
@@ -82,11 +84,63 @@ impl Write for Value {
                 }
                 Ok(())
             }
+            Self::Map(m) => {
+                varint::encode(m.len() as u32, buf);
+                if m.is_empty() {
+                    return Ok(());
+                }
+
+                let key_type_code = m.keys().next().unwrap().type_code();
+                let value_type_code = m.values().next().unwrap().type_code();
+                buf.put_u8(key_type_code as u8);
+                buf.put_u8(value_type_code as u8);
+                for (key, value) in m {
+                    if key.type_code() != key_type_code {
+                        return Err(ImprintError::SchemaError(
+                            "map keys must have same type".into(),
```

Comment:
nit: I wonder if it makes sense to include the types in the error (same below)

Reply:
29ba52d